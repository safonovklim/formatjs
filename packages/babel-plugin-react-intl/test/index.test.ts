import {join, sep} from 'path';
import * as fs from 'fs';
import {transformFileSync} from '@babel/core';
import plugin from '../src';

function trim(str?: string | null) {
  return String(str).replace(/^\s+|\s+$/, '');
}

const skipOutputTests = [
  '.babelrc',
  '.DS_Store',
  'enforceDescriptions',
  'enforceDefaultMessage',
  'extractSourceLocation',
  'extractFromFormatMessageCall',
  'moduleSourceName',
  'icuSyntax',
  'removeDescriptions',
  'overrideMessageFn',
  'removeDefaultMessage',
  'additionalComponentNames',
];

const fixturesDir = join(__dirname, 'fixtures');
const baseDir = join(__dirname, '..');

describe('emit asserts for: ', () => {
  fs.readdirSync(fixturesDir).map(caseName => {
    if (skipOutputTests.indexOf(caseName) >= 0) return;

    it(`output match: ${caseName}`, () => {
      const fixtureDir = join(fixturesDir, caseName);

      // Ensure messages are deleted
      const actualMessagesPath = join(fixtureDir, 'actual.json');
      if (fs.existsSync(actualMessagesPath)) fs.unlinkSync(actualMessagesPath);

      const {code: actual, metadata} = transform(
        join(fixtureDir, 'actual.js')
      )!;
      expect((metadata as any)['react-intl']).toMatchSnapshot();
      // Check code output
      expect(trim(actual)).toMatchSnapshot();

      // Check message output
      expect(require(join(fixtureDir, 'actual.json'))).toMatchSnapshot();
    });
  });
});

describe('options', () => {
  it('removeDefaultMessage should remove default message', () => {
    const fixtureDir = join(fixturesDir, 'removeDefaultMessage');

    const actual = transform(join(fixtureDir, 'actual.js'), {
      removeDefaultMessage: true,
    })!.code;

    // Check code output
    expect(trim(actual)).toMatchSnapshot();

    // Check message output
    expect(require(join(fixtureDir, 'actual.json'))).toMatchSnapshot();
  });
  it('enforces descriptions when enforceDescriptions=true', () => {
    const fixtureDir = join(fixturesDir, 'enforceDescriptions');
    expect(() =>
      transform(join(fixtureDir, 'actual.js'), {
        enforceDescriptions: true,
      })
    ).toThrow(/Message must have a `description`/);
  });

  it('correctly overrides the id and/or defaultMessage when overrideMessageFn is provided', () => {
    const fixtureDir = join(fixturesDir, 'overrideMessageFn');

    const actual = transform(join(fixtureDir, 'actual.js'), {
      overrideMessageFn: (
        id: string,
        defaultMessage: string,
        description: string
      ) => {
        return {
          id: `HELLO.${id}.${defaultMessage.length}.${typeof description}`,
          SUPER_DEFAULT_MESSAGE: defaultMessage,
          defaultMessage: `bye ${defaultMessage}`,
        };
      },
    })!.code;

    // Check code output
    expect(trim(actual)).toMatchSnapshot();

    // Check message output
    expect(require(join(fixtureDir, 'actual.json'))).toMatchSnapshot();
  });

  it('allows no description when enforceDescription=false', () => {
    const fixtureDir = join(fixturesDir, 'enforceDescriptions');
    expect(() =>
      transform(join(fixtureDir, 'actual.js'), {
        enforceDescriptions: false,
      })
    ).not.toThrow();
  });

  it('allows no description when enforceDefaultMessage=false', () => {
    const fixtureDir = join(fixturesDir, 'enforceDefaultMessage');
    expect(() =>
      transform(join(fixtureDir, 'actual.js'), {
        enforceDefaultMessage: false,
      })
    ).not.toThrow();

    // Check message output
    expect(require(join(fixtureDir, 'actual.json'))).toMatchSnapshot();
  });

  it('removes descriptions when plugin is applied more than once', () => {
    const fixtureDir = join(fixturesDir, 'removeDescriptions');
    expect(() =>
      transform(
        join(fixtureDir, 'actual.js'),
        {
          enforceDescriptions: true,
        },
        {
          multiplePasses: true,
        }
      )
    ).not.toThrow();
  });

  it('respects moduleSourceName', () => {
    const fixtureDir = join(fixturesDir, 'moduleSourceName');
    expect(() =>
      transform(join(fixtureDir, 'actual.js'), {
        moduleSourceName: 'react-i18n',
      })
    ).not.toThrow();

    // Check message output
    expect(require(join(fixtureDir, 'actual.json'))).toMatchSnapshot();
  });

  it('respects extractSourceLocation', () => {
    const fixtureDir = join(fixturesDir, 'extractSourceLocation');
    expect(() =>
      transform(join(fixtureDir, 'actual.js'), {
        extractSourceLocation: true,
      })
    ).not.toThrow();

    // Check message output
    const actualMessages = require(join(fixtureDir, 'actual.json'));
    actualMessages.forEach((msg: any) => {
      msg.file = msg.file.replace(/\/|\\/g, '@@sep@@');
    });
    expect(actualMessages).toMatchSnapshot();
  });

  it('respects extractFromFormatMessageCall', () => {
    const fixtureDir = join(fixturesDir, 'extractFromFormatMessageCall');
    expect(() =>
      transform(join(fixtureDir, 'actual.js'), {
        extractFromFormatMessageCall: true,
      })
    ).not.toThrow();

    // Check message output
    expect(require(join(fixtureDir, 'actual.json'))).toMatchSnapshot();
  });

  it('additionalComponentNames', () => {
    const fixtureDir = join(fixturesDir, 'additionalComponentNames');
    expect(() =>
      transform(join(fixtureDir, 'actual.js'), {
        additionalComponentNames: ['CustomMessage'],
      })
    ).not.toThrow();

    // Check message output
    expect(require(join(fixtureDir, 'actual.json'))).toMatchSnapshot();
  });
});

describe('errors', () => {
  it('Properly throws parse errors', () => {
    const fixtureDir = join(fixturesDir, 'icuSyntax');
    expect(() => transform(join(fixtureDir, 'actual.js'))).toThrow(
      /Expected .* but "\." found/
    );
  });
});

const BASE_OPTIONS = {
  messagesDir: baseDir,
};

let cacheBust = 1;

function transform(
  filePath: string,
  options = {},
  {multiplePasses = false} = {}
) {
  function getPluginConfig() {
    return [
      plugin,
      {
        ...BASE_OPTIONS,
        ...options,
      },
      Date.now() + '' + ++cacheBust,
    ];
  }

  return transformFileSync(filePath, {
    plugins: multiplePasses
      ? [getPluginConfig(), getPluginConfig()]
      : [getPluginConfig()],
  });
}
