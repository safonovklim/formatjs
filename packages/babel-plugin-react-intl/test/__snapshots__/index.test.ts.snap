// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`emit asserts for:  output match: FormattedHTMLMessage 1`] = `
Object {
  "messages": Array [
    Object {
      "defaultMessage": "<h1>Hello World!</h1>",
      "description": "The default message.",
      "id": "foo.bar.baz",
    },
  ],
}
`;

exports[`emit asserts for:  output match: FormattedHTMLMessage 2`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireWildcard(require(\\"react\\"));

var _reactIntl = require(\\"react-intl\\");

function _getRequireWildcardCache() { if (typeof WeakMap !== \\"function\\") return null; var cache = new WeakMap(); _getRequireWildcardCache = function () { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; if (obj != null) { var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

class Foo extends _react.Component {
  render() {
    return _react.default.createElement(_reactIntl.FormattedHTMLMessage, {
      id: \\"foo.bar.baz\\",
      defaultMessage: \\"<h1>Hello World!</h1>\\"
    });
  }

}

exports.default = Foo;"
`;

exports[`emit asserts for:  output match: FormattedHTMLMessage 3`] = `
Array [
  Object {
    "defaultMessage": "<h1>Hello World!</h1>",
    "description": "The default message.",
    "id": "foo.bar.baz",
  },
]
`;

exports[`emit asserts for:  output match: FormattedMessage 1`] = `
Object {
  "messages": Array [
    Object {
      "defaultMessage": "Hello World!",
      "description": "The default message.",
      "id": "foo.bar.baz",
    },
  ],
}
`;

exports[`emit asserts for:  output match: FormattedMessage 2`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireWildcard(require(\\"react\\"));

var _reactIntl = require(\\"react-intl\\");

function _getRequireWildcardCache() { if (typeof WeakMap !== \\"function\\") return null; var cache = new WeakMap(); _getRequireWildcardCache = function () { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; if (obj != null) { var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

class Foo extends _react.Component {
  render() {
    return _react.default.createElement(_reactIntl.FormattedMessage, {
      id: \\"foo.bar.baz\\",
      defaultMessage: \\"Hello World!\\"
    });
  }

}

exports.default = Foo;"
`;

exports[`emit asserts for:  output match: FormattedMessage 3`] = `
Array [
  Object {
    "defaultMessage": "Hello World!",
    "description": "The default message.",
    "id": "foo.bar.baz",
  },
]
`;

exports[`emit asserts for:  output match: defineMessages 1`] = `
Object {
  "messages": Array [
    Object {
      "defaultMessage": "Hello World!",
      "description": "The default message",
      "id": "foo.bar.baz",
    },
    Object {
      "defaultMessage": "Hello Nurse!",
      "description": "Another message",
      "id": "foo.bar.biff",
    },
    Object {
      "defaultMessage": "{count, plural, =0 {ðŸ˜­} one {# kitten} other {# kittens}}",
      "description": "Counts kittens",
      "id": "app.home.kittens",
    },
    Object {
      "defaultMessage": "   Some whitespace   ",
      "description": "Whitespace",
      "id": "trailing.ws",
    },
    Object {
      "defaultMessage": "A quoted value ''{value}'",
      "description": "Escaped apostrophe",
      "id": "escaped.apostrophe",
    },
  ],
}
`;

exports[`emit asserts for:  output match: defineMessages 2`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireWildcard(require(\\"react\\"));

var _reactIntl = require(\\"react-intl\\");

function _getRequireWildcardCache() { if (typeof WeakMap !== \\"function\\") return null; var cache = new WeakMap(); _getRequireWildcardCache = function () { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; if (obj != null) { var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

const msgs = (0, _reactIntl.defineMessages)({
  header: {
    \\"id\\": \\"foo.bar.baz\\",
    \\"defaultMessage\\": \\"Hello World!\\"
  },
  content: {
    \\"id\\": \\"foo.bar.biff\\",
    \\"defaultMessage\\": \\"Hello Nurse!\\"
  },
  kittens: {
    \\"id\\": \\"app.home.kittens\\",
    \\"defaultMessage\\": \\"{count, plural, =0 {\\\\uD83D\\\\uDE2D} one {# kitten} other {# kittens}}\\"
  },
  trailingWhitespace: {
    \\"id\\": \\"trailing.ws\\",
    \\"defaultMessage\\": \\"   Some whitespace   \\"
  },
  escaped: {
    \\"id\\": \\"escaped.apostrophe\\",
    \\"defaultMessage\\": \\"A quoted value ''{value}'\\"
  }
});

class Foo extends _react.Component {
  render() {
    return _react.default.createElement(\\"div\\", null, _react.default.createElement(\\"h1\\", null, _react.default.createElement(_reactIntl.FormattedMessage, msgs.header)), _react.default.createElement(\\"p\\", null, _react.default.createElement(_reactIntl.FormattedMessage, msgs.content)), _react.default.createElement(\\"p\\", null, _react.default.createElement(_reactIntl.FormattedMessage, msgs.kittens)));
  }

}

exports.default = Foo;"
`;

exports[`emit asserts for:  output match: defineMessages 3`] = `
Array [
  Object {
    "defaultMessage": "Hello World!",
    "description": "The default message",
    "id": "foo.bar.baz",
  },
  Object {
    "defaultMessage": "Hello Nurse!",
    "description": "Another message",
    "id": "foo.bar.biff",
  },
  Object {
    "defaultMessage": "{count, plural, =0 {ðŸ˜­} one {# kitten} other {# kittens}}",
    "description": "Counts kittens",
    "id": "app.home.kittens",
  },
  Object {
    "defaultMessage": "   Some whitespace   ",
    "description": "Whitespace",
    "id": "trailing.ws",
  },
  Object {
    "defaultMessage": "A quoted value ''{value}'",
    "description": "Escaped apostrophe",
    "id": "escaped.apostrophe",
  },
]
`;

exports[`emit asserts for:  output match: descriptionsAsObjects 1`] = `
Object {
  "messages": Array [
    Object {
      "defaultMessage": "Hello World!",
      "description": Object {
        "metadata": "Additional metadata content.",
        "text": "Something for the translator.",
      },
      "id": "foo.bar.baz",
    },
  ],
}
`;

exports[`emit asserts for:  output match: descriptionsAsObjects 2`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireWildcard(require(\\"react\\"));

var _reactIntl = require(\\"react-intl\\");

function _getRequireWildcardCache() { if (typeof WeakMap !== \\"function\\") return null; var cache = new WeakMap(); _getRequireWildcardCache = function () { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; if (obj != null) { var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

class Foo extends _react.Component {
  render() {
    return _react.default.createElement(_reactIntl.FormattedMessage, {
      id: \\"foo.bar.baz\\",
      defaultMessage: \\"Hello World!\\"
    });
  }

}

exports.default = Foo;"
`;

exports[`emit asserts for:  output match: descriptionsAsObjects 3`] = `
Array [
  Object {
    "defaultMessage": "Hello World!",
    "description": Object {
      "metadata": "Additional metadata content.",
      "text": "Something for the translator.",
    },
    "id": "foo.bar.baz",
  },
]
`;

exports[`emit asserts for:  output match: formatMessageCall 1`] = `
Object {
  "messages": Array [
    Object {
      "defaultMessage": "bar",
      "description": "baz",
      "id": "foo",
    },
  ],
}
`;

exports[`emit asserts for:  output match: formatMessageCall 2`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireWildcard(require(\\"react\\"));

var _reactIntl = require(\\"react-intl\\");

function _getRequireWildcardCache() { if (typeof WeakMap !== \\"function\\") return null; var cache = new WeakMap(); _getRequireWildcardCache = function () { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; if (obj != null) { var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

const objectPointer = {
  id: 'foo.bar.invalid',
  defaultMessage: 'This cannot be extracted',
  description: 'the plugin only supports inline objects'
};

class Foo extends _react.Component {
  render() {
    const msgs = {
      baz: this.props.intl.formatMessage({
        id: 'foo.bar.baz',
        defaultMessage: 'Hello World!',
        description: 'The default message'
      }),
      biff: this.props.intl.formatMessage({
        id: 'foo.bar.biff',
        defaultMessage: 'Hello Nurse!',
        description: 'Another message'
      }),
      invalid: this.props.intl.formatMessage(objectPointer)
    };
    return _react.default.createElement(\\"div\\", null, _react.default.createElement(\\"h1\\", null, msgs.header), _react.default.createElement(\\"p\\", null, msgs.content), _react.default.createElement(\\"span\\", null, _react.default.createElement(_reactIntl.FormattedMessage, {
      id: \\"foo\\",
      defaultMessage: \\"bar\\"
    })));
  }

}

var _default = (0, _reactIntl.injectIntl)(Foo);

exports.default = _default;"
`;

exports[`emit asserts for:  output match: formatMessageCall 3`] = `
Array [
  Object {
    "defaultMessage": "bar",
    "description": "baz",
    "id": "foo",
  },
]
`;

exports[`options additionalComponentNames 1`] = `
Array [
  Object {
    "defaultMessage": "Hello World!",
    "description": "Greeting to the world",
    "id": "greeting-world",
  },
]
`;

exports[`options allows no description when enforceDefaultMessage=false 1`] = `
Array [
  Object {
    "id": "TRANSLATION_KEY",
  },
  Object {
    "id": "foo.bar.baz",
  },
]
`;

exports[`options correctly overrides the id and/or defaultMessage when overrideMessageFn is provided 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireWildcard(require(\\"react\\"));

var _reactIntl = require(\\"react-intl\\");

function _getRequireWildcardCache() { if (typeof WeakMap !== \\"function\\") return null; var cache = new WeakMap(); _getRequireWildcardCache = function () { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; if (obj != null) { var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

const msgs = (0, _reactIntl.defineMessages)({
  header: {
    \\"id\\": \\"HELLO.foo.bar.baz.12.string\\",
    \\"defaultMessage\\": \\"bye Hello World!\\"
  },
  content: {
    \\"id\\": \\"HELLO.foo.bar.biff.12.object\\",
    \\"defaultMessage\\": \\"bye Hello Nurse!\\"
  }
});

class Foo extends _react.Component {
  render() {
    return _react.default.createElement(\\"div\\", null, _react.default.createElement(\\"h1\\", null, _react.default.createElement(_reactIntl.FormattedMessage, msgs.header)), _react.default.createElement(\\"p\\", null, _react.default.createElement(_reactIntl.FormattedMessage, msgs.content)), _react.default.createElement(_reactIntl.FormattedMessage, {
      id: \\"HELLO.foo.bar.zoo.12.object\\",
      defaultMessage: \\"Hello World!\\"
    }), _react.default.createElement(_reactIntl.FormattedHTMLMessage, {
      id: \\"HELLO.foo.bar.delta.21.string\\",
      defaultMessage: \\"<h1>Hello World!</h1>\\"
    }));
  }

}

exports.default = Foo;"
`;

exports[`options correctly overrides the id and/or defaultMessage when overrideMessageFn is provided 2`] = `
Array [
  Object {
    "SUPER_DEFAULT_MESSAGE": "Hello World!",
    "defaultMessage": "bye Hello World!",
    "description": "The default message",
    "id": "HELLO.foo.bar.baz.12.string",
  },
  Object {
    "SUPER_DEFAULT_MESSAGE": "Hello Nurse!",
    "defaultMessage": "bye Hello Nurse!",
    "description": Object {
      "metadata": "Additional metadata content.",
      "text": "Something for the translator.",
    },
    "id": "HELLO.foo.bar.biff.12.object",
  },
  Object {
    "SUPER_DEFAULT_MESSAGE": "Hello World!",
    "defaultMessage": "bye Hello World!",
    "description": Object {
      "metadata": "Additional metadata content.",
      "text": "Something for the translator. Another description",
    },
    "id": "HELLO.foo.bar.zoo.12.object",
  },
  Object {
    "SUPER_DEFAULT_MESSAGE": "<h1>Hello World!</h1>",
    "defaultMessage": "bye <h1>Hello World!</h1>",
    "description": "The default message.",
    "id": "HELLO.foo.bar.delta.21.string",
  },
]
`;

exports[`options removeDefaultMessage should remove default message 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireWildcard(require(\\"react\\"));

var _reactIntl = require(\\"react-intl\\");

function _getRequireWildcardCache() { if (typeof WeakMap !== \\"function\\") return null; var cache = new WeakMap(); _getRequireWildcardCache = function () { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; if (obj != null) { var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

const messages = (0, _reactIntl.defineMessages)({
  foo: {
    \\"id\\": \\"greeting-user\\"
  }
});

class Foo extends _react.Component {
  render() {
    return _react.default.createElement(_reactIntl.FormattedMessage, {
      id: \\"greeting-world\\"
    });
  }

}

exports.default = Foo;"
`;

exports[`options removeDefaultMessage should remove default message 2`] = `
Array [
  Object {
    "defaultMessage": "Hello, {name}",
    "description": "Greeting the user",
    "id": "greeting-user",
  },
  Object {
    "defaultMessage": "Hello World!",
    "description": "Greeting to the world",
    "id": "greeting-world",
  },
]
`;

exports[`options respects extractFromFormatMessageCall 1`] = `
Array [
  Object {
    "defaultMessage": "Hello World!",
    "description": "The default message",
    "id": "foo.bar.baz",
  },
  Object {
    "defaultMessage": "Hello Nurse!",
    "description": "Another message",
    "id": "foo.bar.biff",
  },
  Object {
    "defaultMessage": "bar",
    "description": "baz",
    "id": "foo",
  },
]
`;

exports[`options respects extractSourceLocation 1`] = `
Array [
  Object {
    "defaultMessage": "Hello World!",
    "end": Object {
      "column": 78,
      "line": 6,
    },
    "file": "test@@sep@@fixtures@@sep@@extractSourceLocation@@sep@@actual.js",
    "id": "foo.bar.baz",
    "start": Object {
      "column": 11,
      "line": 6,
    },
  },
]
`;

exports[`options respects moduleSourceName 1`] = `
Array [
  Object {
    "defaultMessage": "Hello World!",
    "description": "The default message.",
    "id": "foo.bar.baz",
  },
]
`;
